orchestrator/services/coordinator_service.py

```
@@ -127,6 +127,26 @@ def _get_power_mode_caps(power_mode: str, db: Session) -> Dict[str, Any]:
+
+    Best-effort by design: this runs on the per-task dispatch path, so a lookup
+    failure (or an unknown stored value) degrades silently to the default rather
+    than failing the task. One indexed workspace read; fine at pilot scale.
+    """
+    try:
+        from core.models.workspaces import Workspace
+
+        ws = db.query(Workspace).filter(Workspace.id == workspace_id).first()
+        mode = (ws.settings or {}).get("power_mode") if ws else None
+        return mode if mode in _POWER_MODE_DEFAULTS else None
+    except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)